### PR TITLE
Rationalize currency unit to "BTC"

### PR DESCRIPTION
**Pull request by rnicoll**
Previously various user-facing strings have used inconsistent currency units "BTC", "btc" and "bitcoins". This adds a single constant and uses it for each reference to the currency unit.

Also adds a description of the unit for --maxtxfee, and adds the missing "amount" field description to the (deprecated) move RPC command.

### CODE REVIEW

**Comment by theochino** [comment]
Hello,

I would suggest using XBT instead of the BTx is the ISO prefix for Bhutan.

Regard,
Theo Chino, NYC
(Participant of the foundation now defunct group for the ISO certification
for Bitcoin)

**Comment by rnicoll** [comment]
I'll leave others to discuss whether it should change, but this patch certainly makes it simpler by using a single constant (although QT wallet values would need to be updated independently)

File: src/init.cpp [hunk 1]
```diff
@@ -342,16 +342,17 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-disablewallet", _("Do not load the wallet and disable wallet RPC calls"));
     strUsage += HelpMessageOpt("-keypool=<n>", strprintf(_("Set key pool size to <n> (default: %u)"), 100));
     if (showDebug)
-        strUsage += HelpMessageOpt("-mintxfee=<amt>", strprintf("Fees (in BTC/Kb) smaller than this are considered zero fee for transaction creation (default: %s)",
```
Comment (luke-jr) [inline src/init.cpp]: At least fix Kb to kb here at the same time..

**Comment by luke-jr** [comment]
@theochino XBT is not a well-defined denomination. Different parts of the community use it for different size bases.

**Comment by luke-jr** [comment]
Also, as to the concept of this PR itself: I don't think we need to define a variable for every reused term. Just changing "btc" and "bitcoins" to "BTC" in literals should be sufficient.

**Comment by rnicoll** [comment]
Fixed kilobyte units, and removed the constant string.

**Comment by rnicoll** [comment]
Just realised these strings are all copied into qt/bitcoinstrings.cpp - are those extracted automatically and I shouldn't touch them, or do they need updating too?

**Comment by laanwj** [comment]
Concept ACK on moving the currency unit to a constant.
Please do leave it as BTC. Changing it to XBT becomes trivial for those that insist on it.

> Just realised these strings are all copied into qt/bitcoinstrings.cpp - are those extracted automatically and I shouldn't touch them, or do they need updating too?

Don't touch the file manually. Translation strings are extracted automatically by `make translate`'s tooling. This updates bitcoinstrings as well as the English ts_file.

**Comment by Diapolo** [comment]
This only seems to cover core code, any reason not to do this for the GUI part?

**Comment by rnicoll** [comment]
@Diapolo Couldn't see any problems in the QT code, apart from the bitcoinstrings.cpp (which are generated), and that code uses its own copy of currency units generally. Do let me know if I've missed anything.

**Comment by laanwj** [comment]
Hm, you removed the parametrization?

**Comment by rnicoll** [comment]
Putting it back at the moment

**Comment by rnicoll** [comment]
Done - let me know if that's okay and I'll squash them

File: src/qt/bitcoinunits.cpp [hunk 2]
```diff
@@ -40,9 +40,9 @@ QString BitcoinUnits::name(int unit)
 {
     switch(unit)
     {
-    case BTC: return QString("BTC");
-    case mBTC: return QString("mBTC");
-    case uBTC: return QString::fromUtf8("μBTC");
+    case BTC: return QString(CURRENCY_UNIT.c_str());
+    case mBTC: return QString(("m" + CURRENCY_UNIT).c_str());
```
Comment (luke-jr) [inline src/qt/bitcoinunits.cpp]: Did you test this? const char * + const char * is going to be a pointer to some undefined memory...

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2278,7 +2279,7 @@ UniValue listunspent(const UniValue& params, bool fHelp)
             "    \"address\" : \"address\",  (string) the bitcoin address\n"
             "    \"account\" : \"account\",  (string) DEPRECATED. The associated account, or \"\" for the default account\n"
             "    \"scriptPubKey\" : \"key\", (string) the script key\n"
-            "    \"amount\" : x.xxx,         (numeric) the transaction amount in btc\n"
+            "    \"amount\" : x.xxx,         (numeric) the transaction amount in BTC\n"
```
Comment (luke-jr) [inline src/wallet/rpcwallet.cpp]: Why not CURRENCY_UNIT here?

File: src/qt/bitcoinunits.cpp [hunk 4]
```diff
@@ -40,9 +40,9 @@ QString BitcoinUnits::name(int unit)
 {
     switch(unit)
     {
-    case BTC: return QString("BTC");
-    case mBTC: return QString("mBTC");
-    case uBTC: return QString::fromUtf8("μBTC");
+    case BTC: return QString(CURRENCY_UNIT.c_str());
```
Comment (laanwj) [inline src/qt/bitcoinunits.cpp]: Let's leave this function as it is. So anyone changing the currency unit will have to change `bitcoinunits` as well, that's OK IMO, that's what it was introduced for.
(This is concatenation is weird, and will mismatch the constant names)

File: src/wallet/rpcwallet.cpp [hunk 5]
```diff
@@ -1686,7 +1687,7 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
             "2. \"includeWatchonly\"    (bool, optional, default=false) Whether to include watchonly addresses in balance calculation and details[]\n"
             "\nResult:\n"
             "{\n"
-            "  \"amount\" : x.xxx,        (numeric) The transaction amount in btc\n"
+            "  \"amount\" : x.xxx,        (numeric) The transaction amount in BTC\n"
```
Comment (laanwj) [inline src/wallet/rpcwallet.cpp]: Seems a few more currency_unit are missing here

Comment (rnicoll) [inline src/wallet/rpcwallet.cpp]: This is what I get for trying to be too clever with search and replace... getting them now.

File: src/qt/bitcoinunits.cpp (hunk 2 above)
Comment (rnicoll) [inline src/qt/bitcoinunits.cpp]: It's std::string + std::string I believe, and certainly seemed to work. Removing as per @laanwj 's request anyway.

Comment (laanwj) [inline src/qt/bitcoinunits.cpp]: No longer relevant here, but for next time: you can use QString::fromStdString to convert a std::string to QString. No need for c_str.

**Comment by rnicoll** [comment]
Fixed, and found a couple of uses of "bitcoin" as a unit too, which I've also fixed.

Thanks for the tip re: QString

File: src/rpcmining.cpp [hunk 6]
```diff
@@ -264,7 +264,7 @@ UniValue getmininginfo(const UniValue& params, bool fHelp)
 }
 
 
-// NOTE: Unlike wallet RPC (which use BTC values), mining RPCs follow GBT (BIP 22) in using satoshi amounts
+// NOTE: Unlike wallet RPC (which use " + CURRENCY_UNIT + " values), mining RPCs follow GBT (BIP 22) in using satoshi amounts
```
Comment (laanwj) [inline src/rpcmining.cpp]: Comment :)

**Comment by laanwj** [comment]
Tested ACK (but please remove the runaway unit parametrization from the comment)

**Comment by rnicoll** [comment]
Fixed and squashed

**Comment by theochino** [comment]
@luke-jr (On your August 1st comment.) - In 3 letters ISO codes, BTC will never be accepted (even it is used today) by the power to be. 

I do prefer the BTC version until I read everything there was on ISO4217 and participated in the foundation working group - https://en.wikipedia.org/wiki/ISO_4217

Once I read it, X (Supranational) BT (Bitcoin) makes more scene than BTC which belong to Bhutan.
I say let make it XBT and make it easy for everyone we want to use the Bitcoin.

Just saying. XBT is free in the ISO code nomenclature. Why not claim it.
People are confused enough as it is ....

**Comment by luke-jr** [comment]
"XBT" is fine as a symbol. My point is that it is not well-defined whether it represents 100,000,000 (BTC), 100,000 (mBTC), 100 (µBTC), or whatever else.

**Comment by theochino** [comment]
I see your point ... leave it as BTC now and wait ... maybe later 1 XBT = 1 µBTC.
The problem, until we solve all the socials ills of the planet, is that will never be a one size fits all solution. Those with wealth will say 1 XBT = 100 BTC, and those on the other side will say 1 XBT = 1 mBTC.

To avoid headaches, I use 1 BTC = 1 XBT, 100'000 mBTC = 100'000 mXBT, and 100 µBTC = 100 µXBT and leave the fixing of the social ill to the pope. ISO standard has a field for the decimal (and I believe it does go to 8) so it would be a "trivial" change to use 1 BTC = 1 XBT makes it simpler in the code to replace.

If the amount changes in the future, we'll change it to something else when we cross that bridge. Anyway, it was a small "loaded" request ....

**Comment by sipa** [comment]
Unifying the currency names across the source code to a single constant is
useful on itself, regardless of what people think that constant should be.

**Pull request merged**